services/sso/src/test/java/software.amazon.awssdk.services.sso/auth/SsoCredentialsProviderTest.java

```diff
@@ -0,0 +1,275 @@
+package software.amazon.awssdk.services.sso.auth;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import javax.xml.bind.DatatypeConverter;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.profiles.ProfileFile;
+import software.amazon.awssdk.services.sso.SsoClient;
+import software.amazon.awssdk.services.sso.internal.ExpiredTokenException;
+import software.amazon.awssdk.services.sso.model.GetRoleCredentialsRequest;
+import software.amazon.awssdk.services.sso.model.GetRoleCredentialsResponse;
+import software.amazon.awssdk.services.sso.model.RoleCredentials;
+import software.amazon.awssdk.utils.StringInputStream;
+import software.amazon.awssdk.utils.Validate;
+
+public class SsoCredentialsProviderTest {
+
+    private static String START_URL = "https://d-abc123.awsapps.com/start-beta";
+
+    private static String testTokenDirectory;
+    private SsoCredentialsProvider provider;
+
+    @Rule
+    public final ExpectedException exception = ExpectedException.none();
+
+    @BeforeClass
+    public static void createTestTokenFile() throws Exception {
+        testTokenDirectory = generateCachedTokenPath(START_URL);
+        createTestTokenFile(testTokenDirectory, "2090-01-01T00:00:00Z");
```

Comment:
CLI generates timestamps in the format `2020-10-27T03:07:06UTC`, does that lead to errors?

Reply:
The "expiresAt" property of the example cached token in the SEP is using "2019-11-14T04:05:45Z", so here this property is parsed by `Instant.parse()`. I checked the javadoc and it looks this method only takes strings in the form of  `DateTimeFormatter.ISO_INSTANT` so I suppose it might lead to errors. Should I add the addressing of the suffix?

Reply:
Thanks 😁